### PR TITLE
SA support and consistent naming for all GVS WDLs

### PR DESCRIPTION
**Pull request by mmorgantaylor**
- `GvsImportGenomes.wdl` - renamed WDL, fixed issue with poorly returned bq load string
- `GvsCreateFilterSet.wdl` - renamed WDL, added SA support, fixed a lot of issues in ExtractFeatures tool surrounding permissions - BQ projectIDs are now being properly passed through, freq_table UDF defined in repo rather than in BQ
- `GvsPrepareCallset.wdl` (done in previous PR)
- `GvsExtractCallset.wdl` - renamed WDL, added SA support
- SA testing README added

all 4 tested with SA in this Terra workspace: https://app.terra.bio/#workspaces/broad-dsp-spec-ops-fc/gvs_sa_testing
testing also without SA in this workspace: https://app.terra.bio/#workspaces/broad-dsp-spec-ops-fc/gvs_testing_no_sa

### CODE REVIEW

File: .dockstore.yml [hunk 1]
```diff
@@ -48,72 +48,76 @@ workflows:
...
+     primaryDescriptorPath: /scripts/variantstore/wdl/GvsCreateFilterSet.wdl
      testParameterFiles:
-       - /scripts/variantstore/wdl/ImportGenomes.example.inputs.json
+       - /scripts/variantstore/wdl/GvsCreateFilterSet.example.inputs.json
      filters:
        branches:
          - master
          - ah_var_store
-   - name: funcotator
+         - mmt_SA_support_and_wdl_renaming
```
Comment (mmorgantaylor) [inline .dockstore.yml:68]: TODO: remove branch references before merging

File: scripts/variantstore/wdl/GvsCreateFilterSet.wdl [hunk 2]
```diff
@@ -343,6 +364,11 @@ task UploadFilterSetToBQ {
 
         export GATK_LOCAL_JAR=~{default="/root/gatk.jar" gatk_override}
 
+        if [ ~{has_service_account_file} = 'true' ]; then
+          gcloud auth activate-service-account --key-file='~{service_account_json}'
```
Comment (kcibul) [inline scripts/variantstore/wdl/GvsCreateFilterSet.wdl:368]: might be good to set the environment variable here as well... even if it's not strictly necessary it keeps out "service account auth" stanza the same everywhere. so our future selves won't wonder why they are different

File: scripts/variantstore/wdl/GvsExtractCallset.wdl [hunk 3]
```diff
@@ -81,45 +84,48 @@ task ExtractTask {
...
             ExtractCohort \
                 --mode GENOMES --ref-version 38 --query-mode LOCAL_SORT \
                 -R "~{reference}" \
-                -O local.vcf.gz \
+                -O ~{output_file} \
                 --local-sort-max-records-in-ram ~{local_sort_max_records_in_ram} \
                 --sample-table ~{fq_sample_table} \
                 --cohort-extract-table ~{fq_cohort_extract_table} \
                 -L ~{intervals} \
+                ~{"-XL " + excluded_intervals} \
```
Comment (kcibul) [inline scripts/variantstore/wdl/GvsExtractCallset.wdl:122]: ❤️

**Review by kcibul — APPROVED** [review]
Looks great -- very thorough

File: .dockstore.yml (hunk 1 above)
Comment (mmorgantaylor) [reply .dockstore.yml:68]: actually will wait for a future PR so it doesn't break for anyone using these WDLs

File: scripts/variantstore/wdl/GvsCreateFilterSet.wdl (hunk 2 above)
Comment (mmorgantaylor) [reply scripts/variantstore/wdl/GvsCreateFilterSet.wdl:368]: great idea

**Review by ahaessly — APPROVED** [review]
👍 
just a few questions and items for discussion

File: scripts/variantstore/wdl/GvsExtractCallset.wdl [hunk 4]
```diff
@@ -81,45 +84,48 @@ task ExtractTask {
...
+        fi
+
         df -h
 
         gatk --java-options "-Xmx9g" \
             ExtractCohort \
                 --mode GENOMES --ref-version 38 --query-mode LOCAL_SORT \
                 -R "~{reference}" \
-                -O local.vcf.gz \
+                -O ~{output_file} \
```
[the remark this answers is not in the document]
Comment (mmorgantaylor) [reply scripts/variantstore/wdl/GvsExtractCallset.wdl:117]: i *think* it doesn't matter - the same way you can enter command line string arguments either with or without quotes. but being consistent would be good!

File: scripts/variantstore/wdl/GvsExtractCallsets.example.inputs.json [hunk 5]
```diff
@@ -0,0 +1,17 @@
+{
+  "GvsExtractCallset.reference": "gs://gcp-public-data--broad-references/hg38/v0/Homo_sapiens_assembly38.fasta",
+  "GvsExtractCallset.reference_index": "gs://gcp-public-data--broad-references/hg38/v0/Homo_sapiens_assembly38.fasta.fai",
+  "GvsExtractCallset.reference_dict": "gs://gcp-public-data--broad-references/hg38/v0/Homo_sapiens_assembly38.dict",
+  "GvsExtractCallset.wgs_intervals": "gs://gcp-public-data--broad-references/hg38/v0/wgs_calling_regions.hg38.interval_list",
+  "GvsExtractCallset.scatter_count": 50,
```
[the remark this answers is not in the document]
Comment (mmorgantaylor) [reply scripts/variantstore/wdl/GvsExtractCallsets.example.inputs.json:6]: it is not! it's defaults from what i found. would love to think deliberately about what these should be.

File: scripts/variantstore/wdl/GvsExtractCallsets.example.inputs.json [hunk 6]
```diff
@@ -0,0 +1,17 @@
+{
```
[the remark this answers is not in the document]
Comment (mmorgantaylor) [reply scripts/variantstore/wdl/GvsExtractCallsets.example.inputs.json:1]: great catch! going to make a separate PR to fix this.

File: scripts/variantstore/wdl/GvsImportGenomes.wdl [hunk 7]
```diff
@@ -348,7 +349,9 @@ task CreateImportTsvs {
       ~{for_testing_only}
 
       if [ ~{has_service_account_file} = 'true' ]; then
+        export GOOGLE_APPLICATION_CREDENTIALS=~{service_account_json}
```
[the remark this answers is not in the document]
Comment (mmorgantaylor) [reply scripts/variantstore/wdl/GvsImportGenomes.wdl:352]: thanks for bringing this up, I wondered about this as well - I did test this WDL with this change and using a SA and it worked fine - https://app.terra.bio/#workspaces/broad-dsp-spec-ops-fc/gvs_sa_testing/job_history/78d0610c-256a-46e4-9f81-fbc10cc2be06 - that had the gvcfs in a separate (non-workspace) bucket, but I did keep your `gsutil cp` step in there, which is likely why we didn't encounter a streaming issue?

**Pull request merged**